pkg/discovery/lifecycle.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package discovery
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+)
+
+// ConnectionSuccess is called when a node is connected to
+func (d *Discovery) ConnectionSuccess(address string, id storj.NodeID) {
+	err := d.cache.Put(context.Background(), id, pb.Node{
```

Comment:
I'll look into this more, I actually don't know if it's possible or worth it

Reply:
As far as I know we should pretty much never use `context.Background()` if it can be avoided. It also looks like you are just passing in `ctx` in `ConnectionFailure`. If it doesn't result in issues, I say don't use `context.Background()`